src/any_llm/providers/bedrock/bedrock.py

```
@@ -243,11 +320,30 @@ async def _alist_models(self, **kwargs: Any) -> Sequence[Model]:
+        Built from the same session (and, when applicable, the same bearer-token credentials)
+        as the runtime client, so overrides applied at construction time aren't silently
+        bypassed for model listing and batch operations. An explicit ``control_client=``
+        constructor kwarg always takes precedence.
+        """
+        if self._custom_control_client is not None:
+            return self._custom_control_client
+        if self._boto_session is not None:
+            return self._boto_session.client("bedrock", **self._bedrock_client_kwargs(self.kwargs))
         return boto3.client("bedrock", **self.kwargs)
 
     def _get_s3_client(self) -> Any:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_get_s3_client`

(ANN401)